### PR TITLE
Replace Streamable w/ Writeable in SingleShardRequest and subclasses

### PR DESCRIPTION
**Pull request by martijnvg**
This commit replaces usages of Streamable with Writeable for the
SingleShardRequest / TransportSingleShardAction classes and subclasses of
these classes.

Note that where possible response fields were made final and default
constructors were removed.

Relates to #34389

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/seqno/RetentionLeaseActions.java [hunk 1]
```diff
@@ -44,9 +44,8 @@
 
 import java.io.IOException;
 import java.util.Objects;
-import java.util.function.Supplier;
 
-/**
+/**˜
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/index/seqno/RetentionLeaseActions.java]: nit: extra ˜

**Review by jimczi — APPROVED** [review]
LGTM

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**